### DISCUSSION TITLE
Parsing segments (nodes) from xpath in libyang

### DISCUSSION THREAD

**Discussion by nbiskup-eridan** [CESNET/libyang#2312] [Q&A]
Hi,
I would like to know if there is an existing function in libyang that can parse segments (nodes) from an xpath expression.

For example, given an xpath like:

```
/o-ran-uplane-conf:user-plane-configuration/rx-array-carriers[name='carrier1']/active
```
Is there a function that would allow me to extract the individual nodes (rx-array-carriers, active) and attributes (name='carrier1')?

I appreciate any guidance or suggestions on how to achieve this using libyang, or if there's a recommended approach.

Thank you for your help!

**Comment by michalvasko** [comment]
No, *libyang* does not expose API for XPath parsing. But there are some functions in *sysrepo*, header `sysrepo/xpath.h`. Just note that if you make extensive use of them, it may slow down your program.

**Reply by nbiskup-eridan** [reply]
Thank you for your response, it was helpful.

I have another question: is there similar functionality for extracting the module name from an xpath expression, such as:

```/module-name:some/xpath/leaf```

**Reply by michalvasko** [reply]
Looking at the *sysrepo* XPath functions, I guess you could use `sr_xpath_next_node_with_ns()` and then just parse the module prefix yourself, you should get `module-name:some` from the function.
